src/phys/mod.rs

```diff
@@ -0,0 +1,98 @@
+//! Tydi physical streams.
+
+/// Tydi stream interface complexity level.
+// TODO(johanpel): implement Complexity "version numbering system" in parser
+#[derive(Debug, Clone, PartialEq)]
+pub struct Complexity {
+    num: Vec<usize>
+}
+
+impl Complexity {
+    // TODO(johanpel): See above todo, this needs fixing
+    pub fn default() -> usize {
```

Comment:
There is `Default` trait in `std`. 
```rust
impl Default for Complexity {
    fn default() -> Self {
        Complexity {
            num: vec![0]
        }
    }
}
```
If all fields in your type implement `Default` you can also derive it.

Reply:
When it returns a Complexity it breaks the parser so we should first have a parser that generates a proper Complexity